compiler/ml/ast_mapper_from0.ml

```diff
@@ -295,10 +295,72 @@ end
 module E = struct
   (* Value expressions for the core language *)
 
-  let map sub {pexp_loc = loc; pexp_desc = desc; pexp_attributes = attrs} =
+  let map_jsx_children sub (e : expression) : Pt.jsx_children =
+    let rec visit (e : expression) : Pt.expression list =
+      match e.pexp_desc with
+      | Pexp_construct
+          ({txt = Longident.Lident "::"}, Some {pexp_desc = Pexp_tuple [e1; e2]})
+        ->
+        sub.expr sub e1 :: visit e2
+      | Pexp_construct ({txt = Longident.Lident "[]"}, ext_opt) -> (
+        match ext_opt with
+        | None -> []
+        | Some e -> visit e)
+      | _ -> [sub.expr sub e]
+    in
+    match e.pexp_desc with
+    | Pexp_construct ({txt = Longident.Lident "[]" | Longident.Lident "::"}, _)
+      ->
+      JSXChildrenItems (visit e)
+    | _ -> JSXChildrenSpreading (sub.expr sub e)
+
+  let try_map_jsx_prop (sub : mapper) (lbl : Asttypes.Noloc.arg_label)
+      (e : expression) : Parsetree.jsx_prop option =
+    match (lbl, e) with
+    | Asttypes.Noloc.Labelled "_spreadProps", expr ->
```

Comment:
Should the constant `_spreadProps` be defined somewhere central perhaps, instead of being inlined? Does it matter?

Reply:
It is used 4 times, so seems like worth doing, do you have any place in mind?

Reply:
If there's a place where logic around JSX prop spreading is located, that's probably a good place.

Reply:
Hmm, it is used in the mapping from and to and in Completion.jsx:

![image](https://github.com/user-attachments/assets/3e84efae-e0ec-4cb0-9cce-f662a5b2ec11)

I don't think this really does anything though for completion.